aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -81,7 +81,7 @@ inline bool _dimreduce_return_trivial_no_ident(Tensor &result, const Tensor &sel
     return true;
   }
 
-  if (self.numel() == 0) {
+  if (self.numel() == 0 && self.ndimension() < 2) {
```

Comment:
I think the check of 0 elements should be on the dim. This way it only fails for one dimensional tensors with zero element, or?